Assets/MixedRealityToolkit.Extensions/StateControl/AppState.cs

```diff
@@ -0,0 +1,221 @@
+    public class AppState : BaseExtensionService, IAppState, IAppStateReadOnly, IAppStateReadWrite
+    {
+        public AppState(IMixedRealityServiceRegistrar registrar, string name, uint priority, BaseMixedRealityProfile profile) : base(registrar, name, priority, profile)
+        {
+            appStateProfile = profile as AppStateProfile;
+            this.Name = name;
+        }
+
+        public override void Initialize()
+        {
+            dataSource = (IAppStateData)GameObject.FindObjectOfType(appStateProfile.DataSourceType);
+            if (dataSource == null)
+            {
+                Debug.Log("Couldn't find component of type " + appStateProfile.DataSourceType.Name + " in scene - creating object now.");
+                GameObject dataSourceObject = new GameObject(Name + " DataSource");
+                try
+                {
+                    dataSource = (IAppStateData)dataSourceObject.AddComponent(appStateProfile.DataSourceType);
+                }
+                catch (Exception e)
+                {
+                    Debug.LogError("Exception when adding component of type " + appStateProfile.DataSourceType.Name + " to dataSourceObject");
+                    Debug.LogException(e);
+                }
+            }
+
+            if (dataSource == null)
+            {
+                throw new Exception("Can't proceed without a data source of type " + typeof(IAppStateData).Name + " - defined type " + appStateProfile.DataSourceType + " yielded no results.");
+            }
+
+            // Tell our data source to fire itself up
+            dataSource.OnServiceInitialized(appStateProfile, new List<ISharingAppObject>() { this });
+
+            // Create our state arrays
+            foreach (Type itemStateType in appStateProfile.SynchronizedTypes)
+                dataSource.CreateStateArray(itemStateType);
+
+            Initialized = true;
+        }
+
+        public AppRoleEnum AppRole { get { return appStateProfile.AppRole; } }
+
+        public bool Initialized { get; protected set; }
+
+        public bool Synchronized
+        {
+            get
+            {
+                // We haven't been initialized yet
+                if (dataSource == null)
+                    return false;
+
+                return dataSource.Synchronized;
+            }
+        }
+
+        public bool ReadyToSynchronize
+        {
+            get { return Initialized && readyToSynchronize; }
+            set { readyToSynchronize = value; }
+        }
+
+        public IEnumerable<Type> ItemStateTypes
+        {
+            get
+            {
+                if (!Initialized)
+                    yield break;
+
+                foreach (Type stateType in appStateProfile.SynchronizedTypes)
+                    yield return stateType;
+            }
+        }
+
+        // Source of our sync lists (implementation will be network back end specific)
+        IAppStateData dataSource;
```

Comment:
docfx won't pick up single // comments but will pick up /// summaries. Its probably worth having any comment you've added use the three slashes vs two in order to reach teh widest audience.